src/openapi/streaming/streaming.js

```diff
@@ -523,6 +525,103 @@ Streaming.prototype.disposeSubscription = function(subscription) {
+        }
+    }
+
+    return subscriptionsToRemove;
+}
+
+function unsubscribeSubscriptionByTag(serviceGroup, url, tag, shouldDisposeSubscription) {
+    const subscriptionsToRemove = getSubscriptionsToRemove.call(this, serviceGroup, url, tag);
+
+    let allSubscriptionsReadyResolve;
+    const allSubscriptionsReady = new Promise((resolve) => {
+        allSubscriptionsReadyResolve = resolve;
+    });
+    const onStateChanged = handleSubscriptionReadyForUnsubscribe.bind(this, subscriptionsToRemove, allSubscriptionsReadyResolve);
+
+    for (let i = 0; i < subscriptionsToRemove.length; i++) {
+        const subscription = subscriptionsToRemove[i];
+
+        ignoreSubscriptions[subscription.referenceId] = true;
+        setTimeout(() => delete ignoreSubscriptions[subscription.referenceId], MS_TO_IGNORE_DATA_ON_UNSUBSCRIBED);
+
+        subscription.addStateChangedCallback(onStateChanged);
+        subscription.onUnsubscribeByTagPending();
+
+        if (shouldDisposeSubscription) {
+            subscription.dispose();
+
+            const indexOfSubscription = this.subscriptions.indexOf(subscription);
+            if (indexOfSubscription >= 0) {
+                this.subscriptions.splice(indexOfSubscription, 1);
+            }
+        }
+    }
+
+    allSubscriptionsReady.then(() => {
+        this.transport.delete(serviceGroup, url + '/{contextId}/?Tag={tag}', {
+            contextId: this.contextId,
+            tag,
+        })
+        .catch((response) => log.error(LOG_AREA, 'An error occurred unsubscribing by tag', { response, serviceGroup, url, tag }))
+        .then(() => {
+            for (let i = 0; i < subscriptionsToRemove.length; i++) {
+                const subscription = subscriptionsToRemove[i];
+                subscription.onUnsubscribeByTagComplete();
+                subscription.removeStateChangedCallback(onStateChanged);
+            }
+        });
+    });
+}
+
+/**
+ * Makes all subscriptions stop at the given serviceGroup and url with the given tag (can be restarted)
+ * See {@link saxo.openapi.Streaming#disposeByTag} for permanently stopping subscriptions by tag.
+ *
+ * @param {string} serviceGroup - the serviceGroup of the subscriptions to unsubscribe
+ * @param {string} url - the url of the subscriptions to unsubscribe
+ * @param {string} tag - the tag of the subscriptions to unsubscribe
+ */
+Streaming.prototype.unsubscribeByTag = function(serviceGroup, url, tag) {
+    unsubscribeSubscriptionByTag.call(this, serviceGroup, url, tag, false);
+};
+
+/**
+ * Disposes all subscriptions at the given serviceGroup and url by tag permanently. They will be stopped and not be able to be started.
+ *
+ * @param {string} serviceGroup - the serviceGroup of the subscriptions to unsubscribe
+ * @param {string} url - the url of the subscriptions to unsubscribe
+ * @param {string} tag - the tag of the subscriptions to unsubscribe
+ */
+Streaming.prototype.disposeByTag = function(serviceGroup, url, tag) {

```

Comment:
Rename to _disposeSubscriptionByTag_, to be more specific, and more consistent with existing disposeSubscription which is similar, but for specific subscription instead of group of tagged ones.